### PR TITLE
[signalfxexporter] Updated for upcoming client auth extension changes

### PR DESCRIPTION
**Pull request by gramidt**
**Description:** 
Updated signalfxexporter for upcoming client auth extension changes

**Link to tracking Issue:**
#3507 

**Testing:**
Unit tests

**Documentation:**
Unit tests

### CODE REVIEW

File: exporter/signalfxexporter/correlation/correlation.go [hunk 1]
```diff
@@ -135,10 +129,13 @@ func (cor *Tracker) AddSpans(ctx context.Context, traces pdata.Traces) error {
 }
 
 // Start correlation tracking.
-func (cor *Tracker) start() {
-	if cor != nil && cor.correlation != nil {
-		cor.correlation.Start()
+func (cor *Tracker) Start(_ context.Context, host component.Host) (err error) {
```
Comment (bogdandrutu) [inline exporter/signalfxexporter/correlation/correlation.go:132]: Why exposing the `Start` func?

File: exporter/signalfxexporter/correlation/correlation.go [hunk 2]
```diff
@@ -124,7 +117,8 @@ func (cor *Tracker) AddSpans(ctx context.Context, traces pdata.Traces) error {
 			false,
 			nil,
 			cor.cfg.SyncAttributes)
-		cor.start()
+
+		cor.correlation.Start()
```
Comment (bogdandrutu) [inline exporter/signalfxexporter/correlation/correlation.go:121]: I think this needs ctx and host?

File: exporter/signalfxexporter/correlation/correlation.go (hunk 1 above)
Comment (gramidt) [reply exporter/signalfxexporter/correlation/correlation.go:132]: This is located in a separate package ('correlation') than the SignalFX Factory ('signalfxexporter').

Comment (jpkrohling) [reply exporter/signalfxexporter/correlation/correlation.go:132]: To see if I got this right: basically, the old start was called by the factory function in this source file, and it was OK to be private. Now, it's an appropriate component `Start` func, called by the component factory, located in another package.

**Review by jpkrohling — APPROVED** [review]

File: exporter/signalfxexporter/correlation/correlation.go (hunk 1 above)
Comment (gramidt) [reply exporter/signalfxexporter/correlation/correlation.go:132]: That's correct, @jpkrohling! The original code was interesting as the correlation tracker delayed being started until 'AddSpans' was called once.

**Comment by gramidt** [comment]
Friendly ping @jrcamp @dmitryax. This is the last PR for the contrib repo to prepare it for the upcoming Client Auth Extension feature.

File: exporter/signalfxexporter/correlation/correlation_test.go [hunk 3]
```diff
@@ -33,20 +37,66 @@ func TestTrackerAddSpans(t *testing.T) {
...
 	attr.InsertString("host.name", "localhost")
 
 	// Add empty first, should ignore.
 	tracker.AddSpans(context.Background(), pdata.NewTraces())
-	assert.Nil(t, tracker.correlation)
 	assert.Nil(t, tracker.traceTracker)
 
 	tracker.AddSpans(context.Background(), traces)
 
-	assert.NotNil(t, tracker.correlation, "correlation context should be set")
```
Comment (dmitryax) [inline exporter/signalfxexporter/correlation/correlation_test.go:48]: Why is this line removed?

File: exporter/signalfxexporter/correlation/correlation.go [hunk 4]
```diff
@@ -105,13 +105,6 @@ func (cor *Tracker) AddSpans(ctx context.Context, traces pdata.Traces) error {
 			return
 		}
 
-		var err error
-		cor.correlation, err = newCorrelationClient(cor.cfg, cor.accessToken, cor.params)
```
Comment (dmitryax) [inline exporter/signalfxexporter/correlation/correlation.go:109]: I see that `cor.correlation` is used in the `cor.traceTracker = tracetracker.New(...)` call. Is there another place where `cor.correlation` initialized the call?

File: exporter/signalfxexporter/correlation/correlation_test.go [hunk 5]
```diff
@@ -33,20 +37,66 @@ func TestTrackerAddSpans(t *testing.T) {
...
+				"abcd",
+				component.ExporterCreateParams{
+					Logger: zap.NewNop(),
+				},
+			)
+
+			err := tracker.Start(context.Background(), componenttest.NewNopHost())
+
+			if tt.wantErr {
+				require.Error(t, err)
```
Comment (dmitryax) [inline exporter/signalfxexporter/correlation/correlation_test.go:96]: is there a way to verify that the error is due to invalid TLSSetting?

File: exporter/signalfxexporter/correlation/correlation_test.go (hunk 3 above)
Comment (gramidt) [reply exporter/signalfxexporter/correlation/correlation_test.go:48]: It was moved up and now lives on L42, since it is now set upon tracker.Start(...), and not from 'AddSpans'. tracker.Start(...) is called from exporter/signalfxexporter/factory.go L94

File: exporter/signalfxexporter/correlation/correlation.go (hunk 4 above)
Comment (gramidt) [reply exporter/signalfxexporter/correlation/correlation.go:109]: Yes, it has been moved to L133 as part of Tracker.Start(...). https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3511/files#diff-c5e5fb87029d1d0003cb0256db0ccfbc8e913b3747d4869fed0d23332a7a8b8dR132

File: exporter/signalfxexporter/correlation/correlation_test.go (hunk 5 above)
Comment (gramidt) [reply exporter/signalfxexporter/correlation/correlation_test.go:98]: Just added the ability to verify the error.

File: exporter/signalfxexporter/correlation/correlation.go (hunk 4 above)
Comment (dmitryax) [reply exporter/signalfxexporter/correlation/correlation.go:109]: oh, sorry, missed that line

**Review by dmitryax — APPROVED** [review]
LGTM

**Comment by gramidt** [comment]
Thank you so much for your review, @dmitryax! 

cc @jpkrohling @pavankrish123

File: exporter/signalfxexporter/correlation/correlation_test.go [hunk 6]
```diff
@@ -33,20 +37,71 @@ func TestTrackerAddSpans(t *testing.T) {
...
+				if tt.errMsg != "" {
+					require.Contains(t, err.Error(), tt.errMsg)
```
Comment (gramidt) [inline exporter/signalfxexporter/correlation/correlation_test.go:100]: @dmitryax - I made a mistake with the past check. I switched it to a contains with a base portion of the error, so that the test works operating systems.

**Comment by tigrannajaryan** [comment]
Thanks @gramidt !
